packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```diff
@@ -92,7 +92,7 @@ function ChallengeSubmissionStepper({
     entityType === EntityType.DOCKER_REPO
       ? steps.SELECT_COMMIT
       : steps.SELECT_EVALUATION
-  const [step, setStep] = useState<Step>(initialStep)
+  const [step, setStep] = useState<Step<StepsEnum>>(initialStep)
```

Comment:
This was also driven by [@typescript-eslint/no-unsafe-enum-comparison](https://typescript-eslint.io/rules/no-unsafe-enum-comparison/). In unchanged code, we compare the current `step.id` to members of the `StepsEnum` that is specific to this component.

I added an optional generic argument to `Step` so in this component we can specify that the `id` must be a member of `StepsEnum`, making the comparison safer, and making eslint happy.